Content.Server/Nutrition/EntitySystems/FlavorProfileSystem.cs

```diff
@@ -86,7 +86,7 @@ private HashSet<string> GetFlavorsFromReagents(Solution solution, int desiredAmo
         var flavors = new HashSet<string>();
         foreach (var (reagent, quantity) in solution.GetReagentPrototypes(_prototypeManager))
         {
-            if (toIgnore != null && toIgnore.Contains(reagent.ID))
+            if (toIgnore != null && toIgnore.Any(s => reagent.ID.Contains(s)))
```

Comment:
This looks bugged if you're checking the ID contains any part of toIgnore.

Reply:
The toIgnoreList cotains the following items ["Nutrients", "Proteins", "Vitamins"], because we want to ignore flavors of ingredients which these types.
Now to the change: Lets assume I have an entity called "MobMouse" this entity contains the "UncookedMeatProtein", if we eat this entity we dont want that the flavor of the "UncookedMeatProtein" to be shown, so we check does "UncookedMeatProtein" contain a single element from the to ignore list which is does("Proteins" element), so we want to ignore it.
If we do it the old ways we would check: does the following list ["Nutrients", "Proteins", "Vitamins"] contains the "UncookedMeatProtein"  which is doesnt, so we wouldnt ignore it even though we want to ignore it.
You can check for yourself that this check does what it is supposed to do, by adding any flavour profile to the `MobMouse` entity in the  `Resources/Prototypes/Entities/Mobs/NPCs/animal.yml` and then try eating a raw mouse.